### ISSUE TITLE
spike to investigate and assess effort for #3277 (telemetry and remote config separation)

### ISSUE THREAD

**Issue by chianger** [mozilla-tw/FirefoxLite#3349]
spike for #3277 (telemetry and remote config separation)

**Comment by cnevinc** [comment]
things to consider
1. Looks like the only thing needs to flip is Analytics. Can we move it to the main thread? Previously we did this cause we called `InstanceID.getInstance().deleteInstanceId()`, is it a must?
2. if we enable remote config default, can we also enable Push and Crashlytics? what's the impact if we don't?
3. If we want to make UI components RemoteConfig aware, what should we do? Should we create a new lifecycle owner? Lifecycle always brings complexity, but there must be some benefits other than immediate UI refresh, what are they?
4. When it comes to immediate UI refresh, we want to see https://firebase.google.com/docs/remote-config/propagate-updates-realtime
5. Life(cycle) is hard, do we want to add another remote source(e.g. static host json) to flip the switch?
